### PR TITLE
[DOC] 📚 Document ReaderStore and WriterStore (#313)

### PR DESCRIPTION
**Pull request by CChemin**
Signed-off-by: CChemin <cecile.chemin@insee.fr>

### CODE REVIEW

**Comment by CChemin** [comment]
still some things to change on credentials (remove the way the password is stored for instance)

File: sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java [hunk 1]
```diff
@@ -24,41 +24,181 @@
...
+  /**
+   * Delete the user id in the store.
+   *
+   * @param organizationId
+   */
+  void deleteUser(String id);
 
-  Group createGroup(String appName, Group group);
+  /**
+   * Create the organization in the store. Throw UnsupportedOperationException if the configuration
```
Comment (micedre) [inline sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java:51]: There is a `@throws` annotation which could be useful here

File: sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java [hunk 2]
```diff
@@ -24,41 +24,181 @@
...
+   * @param generatedPassword password to set to the user, cannot be null
+   * @param pcr not used
+   * @param sendModes not used
+   */
+  void reinitPassword(
+      User user, String generatedPassword, PasswordChangeRequest pcr, List<SendMode> sendModes);
+
+  /**
+   * Change the user password from oldPassword to newPassword. OldPassord and newPassword are
+   * considered as raw even though oldPassword is an hash. Throws InvalidPasswordException if
```
Comment (micedre) [inline sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java:185]: > even though oldPassword is an hash 

Do you mean if it is stored as an hash ? I don't think it is important here, what we want is the correct oldPassword not the hash, the storage implementation is not important (depends on the differents store providers)

**Review by micedre — CHANGES_REQUESTED** [review]

File: sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java (hunk 1 above)
Comment (CChemin) [reply sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java:51]: is it not only for checked exceptions ?

Comment (CChemin) [reply sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java:51]: but maybe we do want the same behavior than a checked exception but without having to try catch

Comment (micedre) [reply sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java:51]: not sure : https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#throwstag

> The purpose of the @throws tag is to indicate which exceptions the programmer must catch (for checked exceptions) or might want to catch (for unchecked exceptions).

We don't want to mention every exception possible but here UnsupportedOperationException is a kind of functionality :)

Comment (CChemin) [reply sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java:51]: ok thanks :) I changed it

File: sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java (hunk 2 above)
Comment (CChemin) [reply sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java:185]: I meant that you if you have a password stored as {SHA}c3q3RSeNwMY7E09Ve9oBHw+MVXg= the password to pass to validate the credential should be {SHA}c3q3RSeNwMY7E09Ve9oBHw+MVXg=. But yes it is irrelevant here.

Comment (micedre) [reply sugoi-api-core/src/main/java/fr/insee/sugoi/core/store/WriterStore.java]: this is not normal, even with a password stored as a hash (as it should be) I would expect the oldPassword to pass to be the real value. If not I'm guessing we don't validate it like we should (with ldap we must validate password by a bind operation, attribute value comparison is very very wrong).

**Review by micedre — APPROVED** [review]

**Pull request merged**